### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Vova-Stoliar**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,24 @@
...
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let arrayToString = '';
+    const cloneElements = this.concat();
+    const separatorToString = separator + '';
+
+    for (let i = 1; i < cloneElements.length; i += 2) {
+      cloneElements.splice(i, 0, separatorToString);
+    }
```
Comment (DmytroMykoliv) [inline src/arrayMethodJoin.js]: this task you can solved without methods 
for (let i = 0; i < this.length; i++) {
        arrayToString += this[i] + `${separator}`;

**Review by DmytroMykoliv — APPROVED** [review]
It is working, but according to the condition you mast not use methods

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let arrayToString = '';
+    const cloneElements = [ ...this ];
```
Comment (VinnieJ-2k20) [inline src/arrayMethodJoin.js:10]: You don't mutate `cloneElements`. Why do you need this copy?

**Review by VinnieJ-2k20 — APPROVED** [review]
Good job 👍 
Just don't create unnecessary variables, if you don't have to mutate original array.

**Pull request closed without merging**